### ISSUE TITLE
Strict equality false positive with Type[Any]

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#7823]
When using `--strict-equality`, the following program produces a false positive error, since `Type[Any]` should overlap with all type objects:

```py
from typing import Any
data: Any = []

# Non-overlapping container check (element type: "Type[Any]", container item type: "ABCMeta")
print(type(data) in [dict, list])  # True
```

cc @ilevkivskyi
